pkg/tcpip/transport/tcp/endpoint.go

```diff
@@ -77,6 +77,14 @@ const (
 	SegOverheadFactor = 2
 )
 
+// Connection direction states used for directionState checks in endpoint struct
```

Comment:
Can we type these and add a named zero-value?

```
type connDirectionState int

// Connection direction states used for directionState checks in endpoint struct
// to detect half-closed connection and deliver POLLRDHUP
const (
	connDirectionStateOpen connDirectionState = 0
	// ...
```

Reply:
Done.